WalletWasabi/Extensions/RPCClientExtensions.cs

```diff
@@ -285,5 +285,66 @@ public static async Task<ISet<uint256>> GetAllDependentsAsync(this IRPCClient rp
 
 			return hashSet;
 		}
+
+		public static async Task<VerboseBlockInfo> GetVerboseBlockAsync(this IRPCClient rpc, uint256 blockId)
+		{
+			var resp = await rpc.SendCommandAsync(RPCOperations.getblock, blockId, 3).ConfigureAwait(false);
+			var blockInfoStr = resp.Result.ToString();
+			var blockInfoJson = JObject.Parse(blockInfoStr);
+			var previousblockhash = blockInfoJson.Value<string>("previousblockhash");
+			var blockInfo = new VerboseBlockInfo()
+			{
+				Hash = uint256.Parse(blockInfoJson.Value<string>("hash")),
+				PrevBlockHash = previousblockhash is { } ? uint256.Parse(previousblockhash) : uint256.Zero,
+				Confirmations = blockInfoJson.Value<ulong>("confirmations"),
+				Height = blockInfoJson.Value<ulong>("height"),
+				BlockTime = Utils.UnixTimeToDateTime(blockInfoJson.Value<uint>("time"))
+			};
+
+			foreach (var txJson in blockInfoJson["tx"])
+			{
+				var tx = new VerboseTransactionInfo()
+				{
+					Id = uint256.Parse(txJson.Value<string>("txid"))
+				};
+
+				var txinJsonInputs = txJson["vin"]
+					.Where(x => !(x["coinbase"] is { }))
+					.Where(x => x["prevout"]["scriptPubKey"].Value<string>("type") == "witness_v0_keyhash");
```

Comment:
You shouldn't do filtering at RPC call level.

Reply:
This is to avoid CodeScene failures.
Or you mean put the code in the `foreach` line? (if yes this will bulk the foreach condition IMO)

Reply:
No. I meant don't do the filtering for `witness_v0_keyhash` in the RPC call itself.

Reply:
Nor to coinbase.

Reply:
Ah right I tend to agree, this was the code review comments form @lontivero (which makes some sense too). I am happy either way.

Reply:
We talked about that we shouldn't do performance improvements in during the first draft, but leave them for later.

Reply:
Done